file.cpp: prepend @TERMUX_PREFIX@ to /tmp path

hashtree.ipp: https://github.com/MoganLab/lolly/pull/346

--- a/System/Files/file.cpp
+++ b/System/Files/file.cpp
@@ -290,7 +290,7 @@
 #if defined(OS_WIN) || defined(OS_MINGW)
   url main_tmp_dir= url_system ("$TMP") * url (".lolly");
 #else
-  url main_tmp_dir= url_system ("/tmp") * url (".lolly");
+  url main_tmp_dir= url_system ("@TERMUX_PREFIX@/tmp") * url (".lolly");
 #endif
   static url tmp_dir= main_tmp_dir * url (as_string (get_process_id ()));
   return (tmp_dir);
--- a/Kernel/Containers/hashtree.ipp
+++ b/Kernel/Containers/hashtree.ipp
@@ -102,7 +102,7 @@ hashtree<K, V>::operator->(void) {
 template <class K, class V>
 inline hashtree<K, V>
 hashtree<K, V>::operator[] (K key) {
-  if (*this->contains (key)) return *this->children (key);
+  if ((*this)->contains (key)) return (*this)->children (key);
   else TM_FAILED ("read-access to non-existent node requested");
 }
 
